core/scale/collection.hpp

```diff
@@ -61,11 +86,7 @@ namespace kagome::scale::collection {
 
     BigInteger required_bytes = collection_size * sizeof(T);
     if (required_bytes > std::numeric_limits<uint64_t>::max()) {
-      return DecodeError::kTooManyItems;
-    }
-
-    if (!stream.hasMore(required_bytes.convert_to<uint64_t>())) {
-      return DecodeError::kNotEnoughData;
+      return DecodeError::TOO_MANY_ITEMS;
     }
```

Comment:
why you removed this check?

Reply:
This check only accelerates negative result in case of simple types. When you encode collection of complex types, this check becomes source of errors. Better remove it at all.